storage/oasis/nodeapi/enigma/node.go

```diff
@@ -0,0 +1,217 @@
+package enigma
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+
+	// nexus-internal data types.
+	"github.com/oasisprotocol/oasis-core/go/common"
+
+	beacon "github.com/oasisprotocol/nexus/coreapi/v22.2.11/beacon/api"
+	consensus "github.com/oasisprotocol/nexus/coreapi/v22.2.11/consensus/api"
+	consensusTx "github.com/oasisprotocol/nexus/coreapi/v22.2.11/consensus/api/transaction"
+	genesis "github.com/oasisprotocol/nexus/coreapi/v22.2.11/genesis/api"
+	governance "github.com/oasisprotocol/nexus/coreapi/v22.2.11/governance/api"
+	scheduler "github.com/oasisprotocol/nexus/coreapi/v22.2.11/scheduler/api"
+
+	"github.com/oasisprotocol/nexus/log"
+	"github.com/oasisprotocol/nexus/storage/oasis/nodeapi"
+
+	// data types for Enigma gRPC APIs.
+	beaconEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/beacon/api"
+	consensusEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/consensus/api"
+	txResultsEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/consensus/api/transaction/results"
+	genesisEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/genesis/api"
+	governanceEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/governance/api"
+	registryEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/registry/api"
+	roothashEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/roothash/api"
+	schedulerEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/scheduler/api"
+	stakingEnigma "github.com/oasisprotocol/nexus/coreapi/v23.0/staking/api"
+)
+
+// EnigmaConsensusApiLite provides low-level access to the consensus API of a
+// Enigma node. To be able to use the old gRPC API, this struct uses gRPC
+// directly, skipping the convenience wrappers provided by oasis-core.
+type EnigmaConsensusApiLite struct {
```

Comment:
Wouldn't it make more sense for this to just be `ConsensusApiLight`. Since enigma is already the package name, it's likely always used as `enigma.ConsensusApiLite`?

Reply:
I think the linter with its default agrees with you :|
I'm not a fan of the rule, but open to renaming this at some point in the way you're suggesting. Right now though, we've already got `CobaltConsensusApiLite` and `DamaskConsensusApiLite` so IMO the higher priority is to keep local consistency.
I'll try to remember when I'm renaming Enigma to whatever E name we come up with.